subsys/net/ip/net_mgmt.c

```diff
@@ -150,7 +150,14 @@ static inline void mgmt_rebuild_global_event_mask(void)
 
 static inline bool mgmt_is_event_handled(u32_t mgmt_event)
 {
-	return ((mgmt_event & global_event_mask) == mgmt_event);
+	return (((NET_MGMT_GET_LAYER(mgmt_event) &
+		  NET_MGMT_GET_LAYER(global_event_mask)) ==
+		 NET_MGMT_GET_LAYER(mgmt_event)) &&
+		((NET_MGMT_GET_LAYER_CODE(mgmt_event) &
+		  NET_MGMT_GET_LAYER_CODE(global_event_mask)) ==
+		 NET_MGMT_GET_LAYER_CODE(mgmt_event)) &&
+		(NET_MGMT_GET_COMMAND(mgmt_event) &
+		 NET_MGMT_GET_COMMAND(global_event_mask)));
```

Comment:
Don't you need to check this for `== NET_MGMT_GET_COMMAND(mgmt_event)` too?

If any bits are set in mgmt_event's command and unset in the global event mask, then net_mgmt_add_event_callback() was never called with a cb->event_mask containing mgmt_event.